pkg/customresourcestate/config.go

```diff
@@ -158,21 +166,47 @@ type ConfigDecoder interface {
 	Decode(v interface{}) (err error)
 }
 
-// FromConfig decodes a configuration source into a slice of customresource.RegistryFactory that are ready to use.
-func FromConfig(decoder ConfigDecoder) ([]customresource.RegistryFactory, error) {
-	var crconfig Metrics
+// FromConfig decodes a configuration source into a slice of `customresource.RegistryFactory` that are ready to use.
+func FromConfig(decoder ConfigDecoder, gvkMaps *discovery.GVKMap) ([]customresource.RegistryFactory, error) {
+	var customResourceConfig Metrics
 	var factories []customresource.RegistryFactory
 	factoriesIndex := map[string]bool{}
-	if err := decoder.Decode(&crconfig); err != nil {
+	if err := decoder.Decode(&customResourceConfig); err != nil {
 		return nil, fmt.Errorf("failed to parse Custom Resource State metrics: %w", err)
 	}
-	for _, resource := range crconfig.Spec.Resources {
+	// TODO: Resolution needs to happen every `discovery.Interval` seconds as well.
+	// Cannot proceed ahead without an instantiated GVKMap object (only for the
+	// first time, otherwise the cache will come up empty on the first run (in the
+	// <`discovery.discoveryInterval` timespan).
+	for {
+		if gvkMaps.Map != nil {
+			break
+		}
+	}
+	var resources []Resource
+	for _, resource := range customResourceConfig.Spec.Resources {
+		gotGVKs := gvkMaps.ResolveGVK(discovery.GroupVersionKind(resource.GroupVersionKind))
+		if gotGVKs != nil {
+			var newResourceList []Resource
+			resource.wasResolved = true
+			for _, gvk := range gotGVKs {
+				newResource := resource
+				newResource.GroupVersionKind = GroupVersionKind(gvk)
+				newResourceList = append(newResourceList, newResource)
+			}
+			resources = append(resources, newResourceList...)
+		} else {
+			resources = append(resources, resource)
+		}
```

Comment:
nil should mean that there is an error we most likely don't want to add the resource in that case. I think in that logic we can handle both partial and complete resources in the same way as long as `ResolveGVK` return an array of 1 when the Resource is complete